src/main/java/hudson/plugins/buildblocker/BlockingJobsMonitor.java

```diff
@@ -147,20 +146,22 @@ private SubTask checkComputerForRunningBuilds(Computer computer) {
         return null;
     }
 
-    public SubTask checkNodeForRunningBuilds(Node node) {
+    public Job checkNodeForRunningBuilds(Node node) {
         if (node == null) {
             return null;
         }
         return checkComputerForRunningBuilds(node.toComputer());
     }
 
-    private SubTask checkForPlannedBuilds(Queue.Item item, List<? extends Queue.Item> buildableItems) {
+    private Job checkForPlannedBuilds(Queue.Item item, List<? extends Queue.Item> buildableItems) {
         for (Queue.Item buildableItem : buildableItems) {
             if (item != buildableItem) {
                 for (String blockingJob : this.blockingJobs) {
-                    AbstractProject project = (AbstractProject) buildableItem.task;
-                    if (project.getFullName().matches(blockingJob)) {
-                        return project;
+                    if (buildableItem.task instanceof Job) {
```

Comment:
Note that for real Workflow compatibility (not just to avoid exceptions) you would want to call `SubTask.getOwnerTask`.